### PR TITLE
Use Lombok @Log everywhere

### PR DESCRIPTION
**Pull request by ssoloff**
## Overview

Follow-up to #3715.

Replaces all static `Logger` fields with Lombok `@Log`.  There was one instance that couldn't be replaced because it was located in a default interface method, and `@Log` is not applicable to interfaces nor methods (`PropertyReader#readIntegerPropertyOrDefault()`).

## Functional Changes

None.

## Refactoring Changes

* If one of the affected `log(Level, String)` calls could be replaced with a shorter call (e.g. `warning()`, `fine()` etc.), I did so.

## Manual Testing Performed

None.

### CODE REVIEW

File: game-core/src/main/java/games/strategy/net/nio/NioReader.java [hunk 1]
```diff
@@ -100,7 +101,7 @@ private void loop() {
               final boolean done = packet.read(channel);
               if (done) {
                 totalBytes += packet.size();
-                if (logger.isLoggable(Level.FINE)) {
+                if (log.isLoggable(Level.FINE)) {
                   final Socket s = channel.socket();
```
Comment (RoiEXLab) [inline game-core/src/main/java/games/strategy/net/nio/NioReader.java]: Why is network code dependent on a logger configuration ?!?!

File: game-core/src/main/java/games/strategy/net/nio/NioWriter.java [hunk 2]
```diff
@@ -67,16 +68,16 @@ private void addNewSocketsToSelector() {
...
   }
 
   private void loop() {
     while (running) {
       try {
-        if (logger.isLoggable(Level.FINEST)) {
-          logger.finest("selecting...");
+        if (log.isLoggable(Level.FINEST)) {
+          log.finest("selecting...");
         }
```
Comment (RoiEXLab) [inline game-core/src/main/java/games/strategy/net/nio/NioWriter.java]: This if check should be redundant IMO

File: game-core/src/main/java/games/strategy/net/nio/NioWriter.java [hunk 3]
```diff
@@ -97,13 +98,13 @@ private void loop() {
             final SocketWriteData packet = getData(channel);
             if (packet != null) {
               try {
-                if (logger.isLoggable(Level.FINEST)) {
-                  logger.finest("writing packet:" + packet + " to:" + channel.socket().getRemoteSocketAddress());
+                if (log.isLoggable(Level.FINEST)) {
+                  log.finest("writing packet:" + packet + " to:" + channel.socket().getRemoteSocketAddress());
                 }
```
Comment (RoiEXLab) [inline game-core/src/main/java/games/strategy/net/nio/NioWriter.java]: This if check should be redundant IMO

File: game-core/src/main/java/games/strategy/net/nio/NioWriter.java [hunk 4]
```diff
@@ -85,8 +86,8 @@ private void loop() {
         // select any new sockets that can be written to
         addNewSocketsToSelector();
         final Set<SelectionKey> selected = selector.selectedKeys();
-        if (logger.isLoggable(Level.FINEST)) {
-          logger.finest("selected:" + selected.size());
+        if (log.isLoggable(Level.FINEST)) {
+          log.finest("selected:" + selected.size());
         }
```
Comment (RoiEXLab) [inline game-core/src/main/java/games/strategy/net/nio/NioWriter.java]: This if check should be redundant IMO

File: game-core/src/main/java/games/strategy/net/nio/NioWriter.java [hunk 5]
```diff
@@ -97,13 +98,13 @@ private void loop() {
...
-                if (logger.isLoggable(Level.FINEST)) {
-                  logger.finest("writing packet:" + packet + " to:" + channel.socket().getRemoteSocketAddress());
+                if (log.isLoggable(Level.FINEST)) {
+                  log.finest("writing packet:" + packet + " to:" + channel.socket().getRemoteSocketAddress());
                 }
                 final boolean done = packet.write(channel);
                 if (done) {
                   totalBytes += packet.size();
-                  if (logger.isLoggable(Level.FINE)) {
+                  if (log.isLoggable(Level.FINE)) {
```
Comment (RoiEXLab) [inline game-core/src/main/java/games/strategy/net/nio/NioWriter.java]: Here again: Why does this even exist?!
This expression should be false for almost all the time.

File: game-core/src/main/java/games/strategy/net/nio/SocketReadData.java [hunk 6]
```diff
@@ -54,8 +55,8 @@ public boolean read(final SocketChannel channel) throws IOException {
         sizeBuffer = ByteBuffer.allocate(4);
       }
       final int size = channel.read(sizeBuffer);
-      if (logger.isLoggable(Level.FINEST)) {
-        logger.finest("read size_buffer bytes:" + size);
+      if (log.isLoggable(Level.FINEST)) {
+        log.finest("read size_buffer bytes:" + size);
       }
```
Comment (RoiEXLab) [inline game-core/src/main/java/games/strategy/net/nio/SocketReadData.java]: This if check should be redundant IMO

File: game-core/src/main/java/games/strategy/net/nio/SocketReadData.java [hunk 7]
```diff
@@ -81,8 +82,8 @@ public boolean read(final SocketChannel channel) throws IOException {
     }
     // http://javaalmanac.com/egs/java.nio/DetectClosed.html
     final int size = channel.read(contentBuffer);
-    if (logger.isLoggable(Level.FINEST)) {
-      logger.finest("read content bytes:" + size);
+    if (log.isLoggable(Level.FINEST)) {
+      log.finest("read content bytes:" + size);
     }
```
Comment (RoiEXLab) [inline game-core/src/main/java/games/strategy/net/nio/SocketReadData.java]: This if check should be redundant IMO

File: game-core/src/main/java/games/strategy/net/nio/SocketWriteData.java [hunk 8]
```diff
@@ -59,8 +60,8 @@ boolean write(final SocketChannel channel) throws IOException {
       if (count == -1) {
         throw new IOException("triplea: end of stream detected");
       }
-      if (logger.isLoggable(Level.FINEST)) {
-        logger.finest("wrote size_buffer bytes:" + count);
+      if (log.isLoggable(Level.FINEST)) {
+        log.finest("wrote size_buffer bytes:" + count);
       }
```
Comment (RoiEXLab) [inline game-core/src/main/java/games/strategy/net/nio/SocketWriteData.java]: This if check should be redundant IMO

File: game-core/src/main/java/games/strategy/net/nio/SocketWriteData.java [hunk 9]
```diff
@@ -71,8 +72,8 @@ boolean write(final SocketChannel channel) throws IOException {
     if (count == -1) {
       throw new IOException("triplea: end of stream detected");
     }
-    if (logger.isLoggable(Level.FINEST)) {
-      logger.finest("wrote content bytes:" + count);
+    if (log.isLoggable(Level.FINEST)) {
+      log.finest("wrote content bytes:" + count);
     }
```
Comment (RoiEXLab) [inline game-core/src/main/java/games/strategy/net/nio/SocketWriteData.java]: This if check should be redundant IMO

**Comment by ssoloff** [comment]
@RoiEXLab Agreed that there are some redundant if checks.  I'll take a look at those tomorrow.  Assigning to myself in the meantime.

**Comment by ssoloff** [comment]
c3843f9ae removes the redundant checks.  In the affected cases, I replaced logger calls that accepted a `String` with the overload that accepts a `Supplier<String>` to avoid string concatenation unless the log level is high enough.  This should preserve the prior behavior.

File: game-core/src/main/java/games/strategy/net/ServerMessenger.java [hunk 10]
```diff
@@ -461,8 +461,8 @@ private void forwardBroadcast(final MessageHeader msg) {
     }
     final SocketChannel fromChannel = nodeToChannel.get(msg.getFrom());
     final List<SocketChannel> nodes = new ArrayList<>(nodeToChannel.values());
-    if (logger.isLoggable(Level.FINEST)) {
-      logger.log(Level.FINEST, "broadcasting to" + nodes);
+    if (log.isLoggable(Level.FINEST)) {
+      log.finest("broadcasting to" + nodes);
     }
```
Comment (RoiEXLab) [inline game-core/src/main/java/games/strategy/net/ServerMessenger.java]: Looks like this case has been missed.

**Pull request merged**